### DISCUSSION TITLE
How can we turn off the heading options in the rich text editor?

### DISCUSSION THREAD

**Discussion by Optiq01** [payloadcms/payload#334] [Q&A]
I have my own custom font sizes which I want to integrate into the rich text editor. I see from the tutorial how I can easily add that functionality as a leaf, however I don't know know if there's a way to toggle off the heading selectors as they won't be needed.  Is there a way to modify this or do I need to just leave them there and ignore them?

**Comment by jmikrut** [comment]
Hey @Optiq01 — great question!  This is certainly doable.

You can explicitly define which types of Slate elements and leaves you'd like to allow on a field by field basis.

Here is a config that only allows the `h3`, `h4`, `ul`, `link`, and `upload` elements, as well as the `bold` and `italic` leaves: 

```js
const field =   {
  name: 'richTextExample',
  type: 'richText',
  admin: {
    elements: [
      'h3',
      'h4',
      'ul',
      'link',
      'upload',
    ],
    leaves: [
      'bold',
      'italic',
    ],
  },
},
``` 

This is also where you'd pass in your own custom elements or leaves. Think of these arrays as the "whitelist" of which elements / leaves are allowed. 

Here is a screenshot of one of our admin panels that features a paired down list of supported elements and leaves, as well as a custom Leaf at the end (the last one, it sets the text to being 50% opaque for UI reasons):

<img width="1673" alt="Screen Shot 2021-10-04 at 7 04 51 PM" src="https://user-images.githubusercontent.com/6651896/135936148-d47e3613-cfd8-42b0-8070-91fd98a89b01.png">

Lastly, [here is the documentation](https://payloadcms.com/docs/fields/rich-text#specifying-which-elements-and-leaves-to-allow) related to this stuff.

Did this answer your question?
